blockchain/stake/staketx.go

```diff
@@ -234,6 +234,48 @@ func isNullFraudProof(tx *wire.MsgTx) bool {
 	return true
 }
 
+// isNullDataScript returns whether or not the passed script is a null
+// data script.
+//
+// NOTE: This function is only valid for version 0 scripts.  It will always
+// return false for other script versions.
+func IsNullDataScript(scriptVersion uint16, script []byte) bool {
+	// The only supported script version is 0.
+	if scriptVersion != 0 {
+		return false
+	}
+
+	// A null script is of the form:
+	//  OP_RETURN <optional data>
+	//
+	// Thus, it can either be a single OP_RETURN or an OP_RETURN followed by a
+	// data push up to MaxDataCarrierSize bytes.
+
+	// The script can't possibly be a null data script if it doesn't start
+	// with OP_RETURN.  Fail fast to avoid more work below.
+	if len(script) < 1 || script[0] != txscript.OP_RETURN {
+		return false
+	}
+
+	// Single OP_RETURN.
+	if len(script) == 1 {
+		return true
+	}
+
+	// OP_RETURN followed by data push up to MaxDataCarrierSize bytes.
+	tokenizer := txscript.MakeScriptTokenizer(scriptVersion, script[1:])
+	if !tokenizer.Next() {
```

Comment:
I would prefer this match what is in `txscript` already.  This reads correctly, but I'd rather have less chance of mistake by just leaving them identical.